crates/madara/client/settlement_client/src/starknet/event.rs

```
@@ -0,0 +1,323 @@
+                                .ok_or_else(|| anyhow::anyhow!("Unable to get block number from event"))
+                                .map(|block_number| CommonMessagingEventData {
+                                    from: event.keys[2].to_bytes_be().to_vec(),
+                                    to: event.keys[3].to_bytes_be().to_vec(),
+                                    selector: event.data[0].to_bytes_be().to_vec(),
+                                    nonce: event.data[1].to_bytes_be().to_vec(),
+                                    payload: {
+                                        let mut payload_array = vec![];
+                                        event.data.iter().skip(3).for_each(|data| {
+                                            payload_array.push(data.to_bytes_be().to_vec());
+                                        });
+                                        payload_array
+                                    },
+                                    fee: None,
+                                    transaction_hash: event.transaction_hash.to_bytes_be().to_vec(),
+                                    message_hash: Some(event.keys[1].to_bytes_be().to_vec()),
+                                    block_number,
+                                    event_index: None,
+                                });
+
+                            Poll::Ready(Some(Some(event_data)))
+                        }
+                        Ok((None, updated_filter)) => {
+                            // Update the filter even when no events are found
+                            self.filter = updated_filter;
+                            Poll::Ready(Some(None))
+                        }
+                        Err(e) => Poll::Ready(Some(Some(Err(e)))),
+                    }
+                }
+                Poll::Pending => Poll::Pending,
+            },
+            None => Poll::Ready(Some(None)), // Handle the case where future is None
```

i think we should throw an error here to inform users something went wrong